### PR TITLE
kmc 3.0.1

### PR DESCRIPTION
**Pull request by tseemann**
- [ ] Have you followed the [guidelines for contributing](https://github.com/brewsci/homebrew-bio/blob/master/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/brewsci/homebrew-bio/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source FORMULA`, where `FORMULA` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict FORMULA` (after doing `brew install FORMULA`)?

-----

### CODE REVIEW

**Comment by tseemann** [comment]
```
clang++ -Wall -O3 -m64 -static-libgcc -static-libstdc++ -fopenmp -pthread -std=c++11  -c kmer_counter/rev_byte.cpp -o kmer_counter/rev_byte.o
In file included from kmer_counter/mem_disk_file.cpp:12:
In file included from kmer_counter/mem_disk_file.h:14:
kmer_counter/defs.h:123:10: fatal error: 'ext/algorithm' file not found
In file included from kmer_counter/rev_byte.cpp:12:
#include <ext/algorithm>
         ^~~~~~~~~~~~~~~
In file included from kmer_counter/rev_byte.h:14:
kmer_counter/defs.h:123:10: fatal error: 'ext/algorithm' file not found
#include <ext/algorithm>
         ^~~~~~~~~~~~~~~
```

I thought the `depends_on gcc` and `fails with clang` would fix this?

File: Formula/kmc.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
+class Kmc < Formula
+  # cite Deorowicz_2015: "https://doi.org/10.1093/bioinformatics/btv022"
+  desc "Fast and frugal disk based k-mer counter"
+  homepage "http://sun.aei.polsl.pl/kmc/"
+  url "https://github.com/marekkokot/KMC/archive/v3.0.1.tar.gz"
+  sha256 "0dbc9254f95541a060d94076d2aa03bb57eb2da114895848f65af0db1e4f8b67"
+  head "https://github.com/marekkokot/KMC.git"
+
+  depends_on "gcc" # needs OpenMP
```
Comment (sjackman) [inline Formula/kmc.rb]:
```ruby
depends_on "gcc" if OS.mac? # needs OpenMP
```

File: Formula/kmc.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  fails_with :clang do
+    build 600
```
Comment (sjackman) [inline Formula/kmc.rb]: Remove this line. It fails with all versions of `clang`. This line says that versions older than 600 fail, but the current version of `clang` is fine.

Comment (tseemann) [reply Formula/kmc.rb]: I originally had removed it, but put it back in in desparation. Removed now.

File: Formula/kmc.rb (hunk 1 above)
Comment (tseemann) [reply Formula/kmc.rb]: Ok. A `needs :openmp` would be handy here :-P

**Comment by tseemann** [comment]
error is out of memory or /tmp space?
i did notice it builds static binaries and took a while to link on my personal machine.
if that is it, would removing `-static` solve some of this.

```
g++-5: internal compiler error: Killed (program cc1plus)
Please submit a full bug report,
with preprocessed source if appropriate.
See <file:///usr/share/doc/gcc-5/README.Bugs> for instructions.
makefile:80: recipe for target 'kmer_counter/kmer_counter.o' failed
make: *** [kmer_counter/kmer_counter.o] Error 4
In file included from kmer_counter/raduls_sse2.cpp:12:0:
kmer_counter/raduls_impl.h:773:1: fatal error: error writing to /tmp/ccfcUZut.s: Cannot allocate memory
 }
 ^
compilation terminated.
makefile:86: recipe for target 'kmer_counter/raduls_sse2.o' failed
make: *** [kmer_counter/raduls_sse2.o] Error 1
```

**Comment by sjackman** [comment]
```
g++-5: internal compiler error: Killed (program cc1plus)
```
It exceeded the 4 GB of memory available on CircleCI.

**Comment by tseemann** [comment]
I've given up for now.   Moved to `depends_on :linux` until someone has time to figure it out.

**Comment by sjackman** [comment]
No worries. That's fine. I'll take a look at macOS some time if I get a chance.

**Comment by tseemann** [comment]
```
==> Installing gcc dependency: binutils
==> Downloading https://linuxbrew.bintray.com/bottles/binutils-2.30.x86_64_linux.bottle.tar.gz
curl: (56) SSL read: error:00000000:lib(0):func(0):reason(0), errno 104
Error: Failed to download resource "binutils"
Download failed: https://linuxbrew.bintray.com/bottles/binutils-2.30.x86_64_linux.bottle.tar.gz
Warning: Bottle installation failed: building from source.
==> Downloading https://ftp.gnu.org/gnu/binutils/binutils-2.30.tar.gz
==> ./configure  --with-sysroot=/ --enable-deterministic-archives --prefix=/home/linuxbrew/.linuxbrew/Cellar/binutils/2.30 --infodir=/home/linuxbrew/.linuxbrew/Cellar/binutils/2.30/share/info --mandir=/home/linuxbrew/.linuxbrew/Cellar/binutils/2.30/share/man --disable-werror --enable-interwork --enable-multilib --enable-64-bit-bfd --enable-gold --enable-plugins --enable-targets=all
==> make
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received
```

**Comment by sjackman** [comment]
Does `export HOMEBREW_FORCE_BREWED_CURL=1` resolve it?

**Comment by sjackman** [comment]
💚 🎉

**Pull request closed without merging**